multiview/cluster/mv_spectral.py

```diff
@@ -0,0 +1,251 @@
+from scipy.spatial.distance import cdist
+from sklearn.base import BaseEstimator
+from sklearn.cluster import KMeans
+from ..utils.utils import check_Xs
+from sklearn.exceptions import NotFittedError
+
+
+class MultiviewSpectralClustering(BaseEstimator):
+
+    '''
+    An implementation of Multi-View Spectral using the
+    basic co-training framework.
+
+    Paramters
+    ---------
+    n_clusters : int
+        The number of clusters
+
+    n_views : int (default=2)
+        The number of different views of data.
+
+    random_state : int (default=None)
+        Determines random number generation for kmeans.
+
+    info_view : int (default=None)
+        The most informative view. Must be between 0 and n_views-1
+        If given, then the final clustering will be performed on the
+        designated view alone. Otherwise, the algorithm will concatenate
+        across all views and cluster on the result.
+
+
+    n_iter : int, optional (default=10)
+        The maximum number of iterations to run the clustering
+        algorithm.
+
+    Attributes
+    ----------
+
+    _n_clusters : int
+        The number of clusters
+
+    _n_views : int
+        The number of different views of the data.
+
+    _random_state : int
+        The seed for the random number generator used during kmeans.
+
+    _info_view : int
+        The index of the most informative view.
+
+    _n_iter : int
+        The number of iterations to run clustering.
+
+    References
+    ----------
+    [1] Abhishek Kumar and Hal Daume. A Co-training Approach for Multiview
+    Spectral Clustering. In International Conference on Machine Learning, 2011
+
+    '''
+
+    def __init__(self, n_clusters, n_views=2, random_state=None,
+                 info_view=None, n_iter=10):
+
+        super().__init__()
+
+        if not (isinstance(n_clusters, int) and n_clusters > 0):
+            msg = 'n_clusters must be a positive integer'
+            raise ValueError(msg)
+
+        if not (isinstance(n_views, int) and n_views > 0):
+            msg = 'n_views must be a positive integer'
+            raise ValueError(msg)
+
+        if random_state is not None:
+            msg = 'random_state must be convertible to 32 bit unsigned integer'
+            try:
+                random_state = int(random_state)
+            except ValueError:
+                raise ValueError(msg)
+            np.random.seed(random_state)
+
+        self._info_view = None
+        if info_view is not None:
+            if (isinstance(info_view, int)
+                    and (info_view >= 0 and info_view < n_views)):
+                self._info_view = info_view
+            else:
+                msg = 'info_view must be an integer between 0 and n_clusters-1'
+                raise ValueError(msg)
+
+        if not (isinstance(n_iter, int) and (n_iter > 0)):
+            msg = 'n_iter must be a positive integer'
+            raise ValueError(msg)
+
+        self._n_clusters = n_clusters
```

Comment:
underscores should be trailing

Reply:
How closely are we following the Scikit-learn API? I just double-checked their docs and it seems that by their conventions, class fields that are set by parameters generally do not have a trailing underscore, while those set by the fit method do have trailing underscores.

Reply:
I guess if that's what they say, up to you. Just leading underscore variables are "hidden" so if someone may want to access them they should probably not have a leading underscore. Up to you